src/python/pants/build_graph/target.py

```diff
@@ -173,11 +174,7 @@ def maybe_readable_identify(cls, targets):
   @classmethod
   def compute_target_id(cls, address):
     """Computes a target id from the given address."""
-    id_candidate = address.path_safe_spec
-    if len(id_candidate) >= 200:
-      # two dots + 79 char head + 79 char tail + 40 char sha1
-      return '{}.{}.{}'.format(id_candidate[:79], sha1(id_candidate).hexdigest(), id_candidate[-79:])
-    return id_candidate
+    return safe_filename(address.path_safe_spec)
```

Comment:
There are two codepaths using `safe_filename`, and the other one was the one that actually caused #5587:
https://github.com/pantsbuild/pants/blob/0bbe913f1c67f298b3191ed51498b20144536888/src/python/pants/invalidation/build_invalidator.py#L214

Lowering this global limit was intended to prevent multiple issues, rather than just in this one consumer.

Reply:
Will expand the description.